examples/Python/ChatApp/app.py

```diff
@@ -0,0 +1,192 @@
+            endpoint=app_config_endpoint,
+            selects=[chat_app_selector],
+            credential=credential,
+            keyvault_credential=credential,  # Use the same credential for Key Vault references
+            trim_prefixes=["ChatApp:"],
+            refresh_on=[WatchKey(key="ChatApp:ChatCompletion:Messages")],
+            on_refresh_success=on_refresh_success,
+        )
+
+        azure_openai_config, azure_client = _create_ai_client(provider, credential)
+
+        # Initialize chat conversation
+        chat_conversation = []
+        print("Chat started! What's on your mind?")
+
+        while True:
+            # Refresh the configuration from Azure App Configuration
+            provider.refresh()
+
+            # Configure chat completion with AI configuration
+            chat_completion_config = _extract_chat_completion_config(provider)
+
+            # Get user input
+            user_input = input("You: ")
+
+            # Exit if user input is empty
+            if not user_input.strip():
+                print("Exiting chat. Goodbye!")
+                break
+
+            # Add user message to chat conversation
+            chat_conversation.append({"role": "user", "content": user_input})
+
+            # Get latest system message from AI configuration
+            chat_messages = _get_chat_messages(chat_completion_config)
+            chat_messages.extend(chat_conversation)
+
+            # Get AI response and add it to chat conversation
+            response = azure_client.chat.completions.create(
+                model=azure_openai_config.deployment_name,
+                messages=chat_messages,
+                max_tokens=chat_completion_config.max_tokens,
+                temperature=chat_completion_config.temperature,
+                top_p=chat_completion_config.top_p,
+            )
+
+            ai_response = response.choices[0].message.content
+            print(f"AI: {ai_response}")
+
+    except Exception as e:
+        print(f"Error: {e}")
+    finally:
+        # Stop the refresher when done
+        if refresher:
+            refresher.stop()
+
+
+def _create_ai_client(
+    azure_openai_config: AzureOpenAIConfiguration, credential: TokenCredential = None
+) -> AzureOpenAI:
+    # Extract Azure OpenAI configuration
+    azure_openai_config = _extract_openai_config(provider)
+    # Create an Azure OpenAI client
+    if azure_openai_config.api_key:
+        return azure_openai_config, AzureOpenAI(
+            azure_endpoint=azure_openai_config.endpoint,
+            api_key=azure_openai_config.api_key,
+            api_version=azure_openai_config.api_version,
+        )
+    token_provider = get_bearer_token_provider(
+        credential or DefaultAzureCredential(),
+        "https://cognitiveservices.azure.com/.default",
+    )
+    return azure_openai_config, AzureOpenAI(
+        azure_endpoint=azure_openai_config.endpoint,
+        azure_ad_token_provider=token_provider,
```

Comment:
Put inline for shorter code.
```suggestion
        azure_ad_token_provider=get_bearer_token_provider(credential, "https://cognitiveservices.azure.com/.default")
```